### ISSUE TITLE
could not connect to bitcoind using json-rpc

### ISSUE THREAD

**Issue by BinaryJava** [ACINQ/eclair#1681]
Eclair gives me these errors when I try to launch through eclair-node.bat. I have the same rpcuser and rpcpassword for both eclair and bitcoin conf files. I have port forwarded incoming and outgoing connections over ipv4 for 9735 and 8333. I have enabled public and private connections to bitcoin core in windows firewall.

```
ERROR fr.acinq.eclair.Boot - fatal error: could not connect to bitcoind using json-rpcfr.acinq.eclair.BitcoinRPCConnectionException: could not connect to bitcoind using json-rpc

ERROR fr.acinq.eclair.Boot - fatal error: C:\Users\kiera\.eclair\eclair.conf: 6: Expecting end of input or a comma, got ':'
```

Here's my bitcoin and eclair conf files:

```
eclair.chain=mainnet
eclair.node-alias=BinaryJava
eclair.bitcoind.rpcport=8332
eclair.server.port=9735
eclair.bitcoind.rpcuser=username
eclair.bitcoind.rpcpassword=password
eclair.bitcoind.zmq=tcp://127.0.0.1:29000
eclair.bitcoind.host=127.0.0.1
eclair.server.public-ips = ["90.194.127.165"]
eclair.node-color=ffa300
eclair.bitcoind.wallet="Lightning wallet"
```

```
testnet=0
server=1
rpcuser=username
rpcpassword=password
rpcport=8332
txindex=1
zmqpubrawblock=tcp://127.0.0.1:29000
zmqpubrawtx=tcp://127.0.0.1:29000
addresstype=p2sh-segwit
```

I am using eclair-node-0.5.0 and running from the .bat file.

Tested on windows 10 with java version "15.0.1"

**Comment by BinaryJava** [comment]
rpcssl=0 needs to be added to bitcoin.conf for it to work.

**Comment by t-bast** [comment]
> addresstype=p2sh-segwit

We'll deprecated this one, I suggest using `bech32` instead.

> rpcssl=0 needs to be added to bitcoin.conf for it to work.

Isn't that because you're using port 8333 instead of 8332 (which is the default)?

**Comment by BinaryJava** [comment]
Default RPC port is 8332 whilst the port for the Bitcoin network is 8333 which I have opened on my router. Is it best to leave addresstype out of the config? It's not mentioned on the readme or wiki. I've seen the rpc ssl has been deprecated for bitcoin core anyway so I've taken that setting out and it still appears to be working.

**Comment by t-bast** [comment]
> Is it best to leave addresstype out of the config?

Yes, that should use bech32 by default on recent versions of bitcoind.

> rpcssl=0 needs to be added to bitcoin.conf for it to work.

That is very strange, I haven't seen anyone run into that before. This should be unnecessary.
Are you sure you're using the right bitcoin core? What version are you running?

**Comment by BinaryJava** [comment]
Bitcoin core v0.21.0. I'm currently reindexing since I added the parameter after I downloaded all the data. I will update once it finishes.

**Comment by t-bast** [comment]
We haven't added support for 0.21.0 yet, we're waiting for the minor bug fix release (0.21.1).

**Comment by BinaryJava** [comment]
Solution was to downgrade bitcoin core version and wait until downloaded blockchain was completely indexed. The bitcoin core wallet does not have GUI to show progress of indexing. Indexing took over 5 days on my system and my indexes folder is currently 29.4gb. Only way I found to track progress was periodically checking size of index folder and to hear when my hard drive was no longer clicking.
